tomplot/local_area_plot.py

```diff
@@ -0,0 +1,35 @@
+import pandas as pd
+
+
+def area_restriction(field_data, coords_X, coords_Y, Y_lim=None, X_lim=None):
+    """
+    a function that takes a data dictionary and will restrict it for a given value
+    For the coordinate fields you must pass in a 1d object so either coords[:,level]
```

Comment:
You're right that the arrays need to be 1D. Should we add some code at the top of the function to check that the arrays are 1D? e.g.

```
if len(np.shape(field_data)) != 1:
    raise ValueError('area_restriction: input data must be 1D to be filtered by pandas data frame')
```